source/hp/fe_collection.cc

```diff
@@ -202,6 +219,53 @@ namespace hp
 
 
 
+  template <int dim, int spacedim>
+  void
+  FECollection<dim, spacedim>::set_hierarchy(
+    const std::function<unsigned int(const unsigned int)> &next,
+    const std::function<unsigned int(const unsigned int)> &prev)
+  {
+    // copy hierarchy functions
+    hierarchy_next = next;
+    hierarchy_prev = prev;
+  }
+
+
+
+  template <int dim, int spacedim>
+  unsigned int
+  FECollection<dim, spacedim>::next_in_hierarchy(
+    const unsigned int fe_index) const
+  {
+    Assert(0 <= fe_index && fe_index < size(),
```

Comment:
here and below, the first half of the condition is implicitly true because you're using an `unsigned` data type